src/Controller/CRUDController.php

```diff
@@ -1067,6 +1070,54 @@ public function getRequest()
         return $this->container->get('request_stack')->getCurrentRequest();
     }
 
+    /**
+     * Contextualize the admin class depends on the current request.
+     *
+     * @throws \RuntimeException
+     */
+    final public function configureAdmin(Request $request): void
+    {
+        $adminCode = $request->get('_sonata_admin');
+
+        if (!$adminCode) {
+            throw new \RuntimeException(sprintf(
```

Comment:
Since the request given in the argument 1 is missing a required parameter, maybe we could use `\InvalidArgumentExcception`.

Reply:
I added it with NEXT_MAJOR comment, since the code is the same it was in `CRUDController::configure`.